github-actions/google-internal-tests/lib/main.ts

```diff
@@ -15,17 +20,32 @@ async function main() {
     );
   }
 
+  const syncBranch = core.getInput('sync-branch', {required: true});
   const githubToken = core.getInput('github-token', {required: true});
   const runTestGuideURL = core.getInput('run-tests-guide-url', {required: false});
   const syncedFilesRaw = core.getInput('synced-files', {required: true});
   const alwaysExternalFilesRaw = core.getInput('always-external-files', {required: false});
 
+  const prNum = context.payload.pull_request!.number;
+  const prHeadSHA = context.payload.pull_request!.head!.sha;
+  const prBaseRef = context.payload.pull_request!.base!.ref;
+
+  if (syncBranch !== prBaseRef) {
+    core.info(`Skipping Google Internal Tests action for PRs not targeting: ${syncBranch}`);
+    return;
+  }
+
   const syncedFiles = constructPatterns(syncedFilesRaw);
   const alwaysExternalFiles = constructPatterns(alwaysExternalFilesRaw);
 
   const github = new Octokit({auth: githubToken});
-  const prNum = context.payload.pull_request!.number;
-  const prHeadSHA = context.payload.pull_request!.head!.sha;
+  const existingGoogleStatus = await findExistingTestStatus(github, prHeadSHA);
+
+  // If there is an existing status already pointing to an internal CL, we do not override
+  // the status. This can happen when e.g. a presubmit-tested PR is closed and reopened.
+  if (existingGoogleStatus && existingGoogleStatus.target_url?.startsWith('http://cl/')) {
```

Comment:
This is the thing I'm a little unhappy about. It relies on the URL we populate from internal. At the same time though this whole action already relies on the context name being `google-internal-tests`- so maybe fine 🤷

Reply:
I view this action as being part of the internal tool (or rather they together make up the tool), so I think its fine to rely on implementation details of the internal piece.